src/meta/src/hummock/compaction_scheduler.rs

```diff
@@ -170,12 +154,23 @@ where
                 .await
             {
                 None => {
-                    tracing::warn!("No idle compactor available.");
+                    let current_compactor_tasks =
+                        self.hummock_manager.list_assigned_tasks_number().await;
+                    tracing::warn!("No idle compactor available. The assigned task number for every compactor is (context_id, count):\n {:?}", current_compactor_tasks);
+                    compact_task.task_status = false;
                     tokio::time::sleep(Duration::from_secs(
```

Comment:
sleep can be moved to line 172?

Reply:
No. Even if we cancel this task success, I do not hope this thread try to get compact task again at once. Because every un-assigned compact task would generate twice IO for ETCD. So sleep is necessary